README.md

```diff
@@ -1,196 +1,289 @@
-
-Logging, Notification and Error Handling
-------------------------------------------
-**Uploading**
-
-A log of the CRON command (executed with `bash -e`) is written to the user's home folder `~/dx-stream_cron.log` and can be used to check the top level command triggered.
-
-The verbose log of the upload process (generated by the top-level `monitor_runs.py`) is written to the user's home folder `~/monitor.log`.
-
-These logs can be used to diagnose failures of upload from the local machine to DNAnexus.
-
-**Downstream applet**
-
-The downstream applet will be run in the project that the RUN directory is uploaded to (as specified in role variable `upload_project`). Users can log in to their DNAnexus account (corresponding to the `dx_token` or `dx_user_token`) and navigate to the upload project to monitor the progress of the applet triggered. Typically, on failure of a DNAnexus job, the user will receive a notification email, which will direct the user to check the log of the failed job for further diagnosis and debugging.
-
-License
--------
-
-Apache
-
-Author Information
-------------------
-
-DNAnexus (email: support@dnanexus.com)
+
+[![Build Status](https://travis-ci.org/dnanexus-rnd/dx-streaming-upload.svg?branch=master)](https://travis-ci.org/dnanexus-rnd/dx-streaming-upload)
+
+dx-streaming-upload
+===================
+
+The dx-streaming-upload Ansible role packages the streaming upload module for increamentally uploading a RUN directory from an Illumina sequencer onto the DNAnexus platform.
+
+Instruments that this module support include the Illumina MiSeq, NextSeq, HiSeq-2500, HiSeq-4000, HiSeq-X and NovaSeq.
+
+
+## Table of Contents
+1. [Dependencies](#dependencies)
+2. [Requirements](#requirements) 
+3. [Installation](#installation)
+4. [Examples](#examples)
+5. [Example workflows](#example-workflows)
+6. [Troubleshooting](#troubleshooting)
+
+## Dependencies
+
+Python 2.7 is needed. This program is not compatible with Python 3.X.
+
+Minimal Ansible version: 2.0.
+
+This program is intended for Ubuntu 14.04 and 16.04, and has been tested on Red Hat 7.4/7.5 and OLE (Oracle Linux Enterprise) 7. It has not been tested on any other versions. 
+
+## Requirements
+
+Users of this module needs a DNAnexus account and its accompanying authentication. To register for a trial account, visit the [DNAnexus homepage](https://platform.dnanexus.com/register).
+
+More information and tutorials about the DNAnexus platform can be found at the [DNAnexus wiki page](https://wiki.dnanexus.com/Home).
+
+The remote-user that the role is run against must possess READ access to monitored_folder and WRITE access to disk for logging and temporary storage of tar files. These are typically stored under the remote-user's home directory, and is specified in the file monitor_run_config.template or as given explicitly by the variables local_tar_directory and local_log_directory.

```

Comment:
Remote-user is a little bit confusing, let's define it and find a better name

Reply:
I have updated this phrase to local user